docs/model_developers/plugins.rst

```diff
@@ -128,3 +108,67 @@ see :ref:`moddev-module`.
 
 .. todo:: context
 .. todo:: new statements
+
+
+
+
+Deprecate plugins
+========================
+
+To deprecate a plugin the :func:`~inmanta.plugins.deprecated` decorator can be used in combination with the :func:`~inmanta.plugins.plugin`
+decorator. Using this decorator will log a warning message when the function is called. This decorator also accepts an
+optional argument ``replaced_by`` which can be used to potentially improve the warning message by telling which other
+plugin should be used in the place of the current one.
+
+for example if the plugin below is called:
+
+.. code-block:: python
+    :linenos:
+
+    from inmanta.plugins import plugin,deprecated
+
+    @deprecated(replaced_by="my_new_plugin")
+    @plugin
+    def printf():
+        """
+            Prints inmanta
+        """
+        print("inmanta")
+
+
+it will give following warning:
+
+.. code-block::
+
+    Plugin 'printf' in module 'inmanta_plugins.<module_name>' is deprecated. It should be replaced by 'my_new_plugin'
+
+Should de replace_by argument be omitted, the warning would look like this:
+
+.. code-block::
+
+    Plugin 'printf' in module 'inmanta_plugins.<module_name>' is deprecated.
+
+If you want your module to stay backwards compatible you will also need to add a little piece of code that changes how
+:func:`~inmanta.plugins.deprecated` is imported as it does not exist in all versions.
+
+the previous example would then look like this:
+
+.. code-block:: python
+    :linenos:
+
+    from inmanta.plugins import plugin
+
+    try:
+        from inmanta.plugins import deprecated
+    except ImportError:
+    deprecated = lambda f=None, **kwargs: f if f is not None else deprecated
```

Comment:
```suggestion
        deprecated = lambda function=None, **kwargs: function if function is not None else deprecated
```
Just in case someone would pass `function` as a key-word argument, make sure the name matches the one from core.